### PR TITLE
refactor(app): replace Text with StyledText

### LINKED ISSUE #11149: App [Housekeeping]: Replacing Text with StyledText
Basically, most components use StyledText (`atoms/text) but still, some components use `Text`. We would need to replace `Text` with `StyledText` for consistency.  Also, the default text color was updated, so the PR for this should remove `color={COLORS.darkBlack}` if `Text` or `StyledText` uses it.

- [ ] replace `Text` with `StyledText`
- [ ] remove `color={COLORS.darkBlack}` if `StyledText` use it

### PR DESCRIPTION
**Pull request by koji**
# Overview
This PR will replace Text with StyledText and remove color={COLORS.darkBlack} which is the current
default text color. The files this PR updated are using components from `atoms` since the files that don't use components from `atoms` will not be needed anymore or will be updated soon.

fix #11149

# Changelog
- Text --> StyledText
- remove color={COLORS.darkBlack}
- do a few housekeeping 
  ex width={'11rem'} --> width="11rem"
  0.5rem --> SPACING.spacing3

# Review requests
- Text --> StyledText
- The changes align with our recent update

# Risk assessment
low

### CODE REVIEW

File: app/src/atoms/Slideout/Slideout.stories.tsx [hunk 1]
```diff
@@ -23,22 +17,25 @@ const Template: Story<React.ComponentProps<typeof Slideout>> = args => (
 
 const Children = (
   <React.Fragment>
-    <Text
+    <StyledText
       fontWeight={600}
```
Comment (jerader) [inline app/src/atoms/Slideout/Slideout.stories.tsx]: maybe we can change this to `TYPOGRAPHY.fontWeightSemiBold ` but I am not sure if you are planning to go through and change all the `fontWeights` in a separate PR

File: app/src/atoms/StatusLabel/index.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const StatusLabel = (props: StatusLabelProps): JSX.Element | null => {
           marginX={SPACING_1}
           data-testid="status_circle"
         >
-          {pulse ? (
+          {pulse ?? false ? (
```
Comment (jerader) [inline app/src/atoms/StatusLabel/index.tsx]: why did you add this `?? false`?

File: app/src/organisms/Devices/RobotOverviewOverflowMenu.tsx [hunk 3]
```diff
@@ -103,7 +103,7 @@ export const RobotOverviewOverflowMenu = (
       />
       {showOverflowMenu ? (
         <Flex
-          width={'12rem'}
+          width="12rem"
           zIndex={10}
           borderRadius={'4px 4px 0px 0px'}
           boxShadow={'0px 1px 3px rgba(0, 0, 0, 0.2)'}
```
Comment (jerader) [inline app/src/organisms/Devices/RobotOverviewOverflowMenu.tsx]:
```suggestion
          boxShadow="0px 1px 3px rgba(0, 0, 0, 0.2)"
```

File: app/src/organisms/Devices/RobotOverviewOverflowMenu.tsx [hunk 4]
```diff
@@ -103,7 +103,7 @@ export const RobotOverviewOverflowMenu = (
       />
       {showOverflowMenu ? (
         <Flex
-          width={'12rem'}
+          width="12rem"
           zIndex={10}
           borderRadius={'4px 4px 0px 0px'}
```
Comment (jerader) [inline app/src/organisms/Devices/RobotOverviewOverflowMenu.tsx]:
```suggestion
          borderRadius="4px 4px 0px 0px"
```

File: app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetailModal.tsx [hunk 5]
```diff
@@ -164,7 +164,7 @@ export const LabwarePositionCheckStepDetailModal = (
                 as={'h5'}
                 position={POSITION_ABSOLUTE}
                 top="25%"
-                marginLeft={'6.7rem'}
+                marginLeft="6.7rem"
                 color={'#ff5b5b'}
```
Comment (jerader) [inline app/src/organisms/LabwarePositionCheck/LabwarePositionCheckStepDetailModal.tsx]:
```suggestion
                color="#ff5b5b"
```

File: app/src/organisms/ProtocolDetails/ProtocolLiquidsDetails.tsx [hunk 6]
```diff
@@ -20,7 +20,7 @@ export const ProtocolLiquidsDetails = (): JSX.Element => {
     <Flex
       css={HIDE_SCROLLBAR}
       flexDirection={DIRECTION_COLUMN}
-      maxHeight={'25rem'}
+      maxHeight="25rem"
       overflowY={'auto'}
```
Comment (jerader) [inline app/src/organisms/ProtocolDetails/ProtocolLiquidsDetails.tsx]:
```suggestion
      overflowY="auto"
```

File: app/src/organisms/ProtocolDetails/OverflowMenu.tsx [hunk 7]
```diff
@@ -75,7 +75,7 @@ export function OverflowMenu(props: OverflowMenuProps): JSX.Element {
       />
       {showOverflowMenu ? (
         <Flex
-          width={'12rem'}
+          width="12rem"
           zIndex={10}
           borderRadius={'4px 4px 0px 0px'}
           boxShadow={'0px 1px 3px rgba(0, 0, 0, 0.2)'}
```
Comment (jerader) [inline app/src/organisms/ProtocolDetails/OverflowMenu.tsx]:
```suggestion
          borderRadius="4px 4px 0px 0px"
          boxShadow="0px 1px 3px rgba(0, 0, 0, 0.2)"
```

File: app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx [hunk 8]
```diff
@@ -227,16 +227,16 @@ function RunSetupLoader(): JSX.Element | null {
...
         />
-        <Text
+        <StyledText
           justifyContent={JUSTIFY_SPACE_BETWEEN}
-          as={'h3'}
+          as="h3"
           color={C_DARK_GRAY}
           marginBottom={SPACING_5}
           fontWeight={FONT_WEIGHT_REGULAR}
           fontSize={FONT_SIZE_BODY_2}
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx]:
```suggestion
          marginBottom="4rem"
          fontWeight={TYPOGRAPHY.fontWeightRegular}
          fontSize={TYPOGRAPHY.fontSizeH3 }
```

File: yarn.lock [hunk 9]
```diff
@@ -6697,7 +6697,7 @@ cli-truncate@^1.1.0:
     slice-ansi "^1.0.0"
     string-width "^2.0.0"
 
-cli-truncate@^2.0.0:
+cli-truncate@^2.1.0:
```
Comment (jerader) [inline yarn.lock]: why are you changing stuff in `yarn.lock`?

**Review by jerader — COMMENTED** [review]
left some comments on stuff you missed. But I also have a few questions:

1. why did you change `yarn.lock`? 
2. why did you change the logic in `StatusLabel` for when to animate the icon?

File: app/src/atoms/StatusLabel/index.tsx (hunk 2 above)
Comment (koji) [reply app/src/atoms/StatusLabel/index.tsx]: avoid the unexpected null issue.
maybe the following is better?
```ts
pulse != null && pulse ? (
```

File: yarn.lock (hunk 9 above)
Comment (koji) [reply yarn.lock]: Oh, I forgot to update `yarn.lcok` This should not be included in this PR.
I will fix this.

File: app/src/atoms/StatusLabel/index.tsx (hunk 2 above)
Comment (jerader) [reply app/src/atoms/StatusLabel/index.tsx]: I like that better, it is more clear what the `false` means

**Comment by koji** [comment]
This PR replaces `{'string'}` with "string". The rest will be solved by  #11211.

File: app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx [hunk 10]
```diff
@@ -135,25 +130,25 @@ export function RunSetupCard(): JSX.Element | null {
...
       width="100%"
-      marginTop={SPACING_3}
-      paddingY={SPACING_3}
+      marginTop={SPACING.spacing4}
+      paddingY={SPACING.spacing4}
       backgroundColor={C_WHITE}
     >
-      <Text
+      <StyledText
         as="h2"
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx:138]: just a random question but how come sometimes `as=""` works, and sometimes it doesn't? Like for example, here we use `as="h2"` but if you were to use `h2Semibold`, you implement it like `css="h2Semibold"`

File: yarn.lock [hunk 11]
```diff
@@ -23208,5 +23208,4 @@ yup@0.32.9:
 
 zwitch@^1.0.0:
   version "1.0.5"
-  resolved "https://registry.yarnpkg.com/zwitch/-/zwitch-1.0.5.tgz#d11d7381ffed16b742f6af7b3f223d5cd9fe9920"
-  integrity sha512-V50KMwwzqJV0NpZIZFwfOD5/lyny3WlSzRiXgA0G7VUnRlqttta1L6UQIHzd6EuBY/cHGfwTIck7w1yH6Q5zUw==
+  resolved "https://registry.yarnpkg.com/zwitch/-/zwitch-1.0.5.tgz#d11d7381ffed16b742f6af7b3f223d5cd9fe9920"
```
Comment (jerader) [inline yarn.lock]: there is still this `yarn.lock` addition. Should it be here?

**Review by jerader — APPROVED** [review]
left a few questions - you still have a change to `yarn.lock` Otherwise, LGTM

File: yarn.lock (hunk 11 above)
Comment (koji) [reply yarn.lock]: This is weird. Actually, I removed this line in [aa59e2f](https://github.com/Opentrons/opentrons/pull/11205/commits/aa59e2fe13b68c9f7a88851d7137d35f5c58b299)

File: app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx (hunk 10 above)
Comment (koji) [reply app/src/organisms/ProtocolSetup/RunSetupCard/index.tsx:138]: >just a random question but how come sometimes as="" works, and sometimes it doesn't? 

I think Brent updated StyledText in May or June to fix `as=""` issue. Do you know any conditions that `as=""` doesn't work properly?

> Like for example, here we use as="h2" but if you were to use h2Semibold, you implement it like css="h2Semibold"

For that, I guess we may need to have a coding rule or code cookbook to unify the way to implement the text part.
In my case, I follow the AppSettings style since I started working on that part.
